src/main/java/com/arpnetworking/metrics/common/sources/CollectdHttpSource.java

```diff
@@ -0,0 +1,276 @@
+     */
+    /* package private */ static final class Actor extends UntypedActor {
+        /**
+         * Creates a {@link Props} for this actor.
+         *
+         * @param source The {@link CollectdHttpSource} to send notifications through.
+         * @return A new {@link Props}
+         */
+        /* package private */ static Props props(final CollectdHttpSource source) {
+            return Props.create(Actor.class, source);
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void onReceive(final Object message) throws Exception {
+            if (message instanceof RequestReply) {
+                final RequestReply request = (RequestReply) message;
+                Source.single((HttpRequest) request.getRequest())
+                        .via(_processGraph)
+                        .toMat(_sink, Keep.right())
+                        .run(_materializer)
+                        .whenComplete((done, err) -> {
+                            if (err != null) {
+                                BAD_REQUEST_LOGGER.warn()
+                                        .setMessage("Error handling collectd post")
+                                        .setThrowable(err)
+                                        .log();
+                                if (err instanceof ParsingException) {
+                                    request.getResponse().complete(HttpResponse.create().withStatus(400));
+                                } else if (err instanceof ConstraintsViolatedException) {
+                                    final ConstraintsViolatedException violationException = (ConstraintsViolatedException) err;
+                                    final StringBuilder errorMessages = new StringBuilder();
+                                    for (final ConstraintViolation violation : violationException.getConstraintViolations()) {
+                                        final String error = violation.getMessage();
+                                        errorMessages.append(error.substring(error.indexOf('_') + 1)).append("\n");
+                                    }
+                                    final HttpResponse response = HttpResponse.create()
+                                            .withStatus(400)
+                                            .withEntity(errorMessages.toString());
+                                    request.getResponse().complete(response);
+                                } else {
+                                    request.getResponse().complete(HttpResponse.create().withStatus(500));
+                                }
+                            } else {
+                                request.getResponse().complete(HttpResponse.create().withStatus(200));
+                            }
+                        });
+            } else {
+                unhandled(message);
+            }
+        }
+
+        /**
+         * Constructor.
```

Comment:
Heh, yeah.  This is a complicated part.  A picture would go a long way in making it understandable.  I'll add one along with a description in comments.  You like ASCII art, right?